air-script/tests/selectors/selectors_combine_complex.rs

```diff
@@ -97,6 +97,6 @@ impl Air for SelectorsAir {
         let main_next = main_frame.next();
         let aux_current = aux_frame.current();
         let aux_next = aux_frame.next();
-        result[0] = ((aux_rand_elements.rand_elements()[0] + E::ONE * aux_rand_elements.rand_elements()[1] + E::from(Felt::new(2_u64)) * aux_rand_elements.rand_elements()[2]) * E::from(main_current[0]) * E::from(main_current[5]) + E::ONE - E::from(main_current[0]) * E::from(main_current[5])) * ((aux_rand_elements.rand_elements()[0] + E::ONE * aux_rand_elements.rand_elements()[1] + E::from(Felt::new(2_u64)) * aux_rand_elements.rand_elements()[2]) * (E::ONE - E::from(main_current[0])) * E::from(main_current[1]) * E::from(main_current[5]) + E::ONE - (E::ONE - E::from(main_current[0])) * E::from(main_current[1]) * E::from(main_current[5])) * aux_current[0] - ((aux_rand_elements.rand_elements()[0] + E::from(Felt::new(3_u64)) * aux_rand_elements.rand_elements()[1] + E::from(Felt::new(4_u64)) * aux_rand_elements.rand_elements()[2]) * (E::ONE - E::from(main_current[0])) * (E::ONE - E::from(main_current[1])) * E::from(main_current[4]) + E::ONE - (E::ONE - E::from(main_current[0])) * (E::ONE - E::from(main_current[1])) * E::from(main_current[4])) * aux_next[0];
+        result[0] = ((aux_rand_elements.rand_elements()[0] + aux_rand_elements.rand_elements()[1] + E::from(Felt::new(2_u64)) * aux_rand_elements.rand_elements()[2]) * E::from(main_current[0]) * E::from(main_current[5]) + E::ONE - E::from(main_current[0]) * E::from(main_current[5])) * ((aux_rand_elements.rand_elements()[0] + aux_rand_elements.rand_elements()[1] + E::from(Felt::new(2_u64)) * aux_rand_elements.rand_elements()[2]) * (E::ONE - E::from(main_current[0])) * E::from(main_current[1]) * E::from(main_current[5]) + E::ONE - (E::ONE - E::from(main_current[0])) * E::from(main_current[1]) * E::from(main_current[5])) * aux_current[0] - ((aux_rand_elements.rand_elements()[0] + E::from(Felt::new(3_u64)) * aux_rand_elements.rand_elements()[1] + E::from(Felt::new(4_u64)) * aux_rand_elements.rand_elements()[2]) * (E::ONE - E::from(main_current[0])) * (E::ONE - E::from(main_current[1])) * E::from(main_current[4]) + E::ONE - (E::ONE - E::from(main_current[0])) * (E::ONE - E::from(main_current[1])) * E::from(main_current[4])) * aux_next[0];
     }
```

Comment:
Above, we have `1-current[0]` twice. Is this something that's easy to combine?

Reply:
We should also be able to compute this (and other similar instances) over the base field, and then cast to the extension field. This should be slightly faster.